src/idl_gen_rust.cpp

```diff
@@ -35,10 +35,18 @@ std::string MakeSnakeCase(const std::string &in) {
   for (size_t i = 0; i < in.length(); i++) {
     if (islower(in[i])) {
       s += static_cast<char>(in[i]);
-    } else {
-      if (i > 0) {
-        s += '_';
+    } else if (i == 0 ) {
+      s += static_cast<char>(tolower(in[i]));
+    }
+    // Prevent underscore duplicates.
+    else if (in[i - 1] != '_') {
```

Comment:
this seems overcomplicated logic. Wouldn't `if(c == '_') { /*ignore*/ } else if(upper(c)) { if(i) s += '_'; s += lower(c); } else s += c;` suffice?

Reply:
@aardappel This would not work against a string which is already in snake case or a string containing a numeric character. But I will make it simpler.